src/main/java/seedu/address/ui/NoteWindow.java

```diff
@@ -0,0 +1,184 @@
+    private static final KeyCombination SAVE_KEY = new KeyCodeCombination(KeyCode.S, CTRL);
+    private static final KeyCombination EXIT_AND_SAVE_KEY = new KeyCodeCombination(KeyCode.Q, CTRL);
+    private static final KeyCombination EXIT_KEY = new KeyCodeCombination(KeyCode.W, CTRL);
+    private static final KeyCombination TIME_STAMP_KEY = new KeyCodeCombination(KeyCode.T, CTRL);
+
+    @FXML
+    private TextArea noteTextArea;
+
+    private final Person person;
+
+    private final Logic logic;
+
+
+    /**
+     * Creates a new NoteWindow.
+     */
+    public NoteWindow(Person person, Logic logic) {
+        super(FXML);
+        noteTextArea.setText(person.getNote().value);
+        this.person = person;
+        this.logic = logic;
+        getRoot().setTitle(person.getName().toString());
+    }
+
+    /**
+     * Show the noteWindow.
+     */
+    public void show() {
+        getRoot().show();
+        getRoot().centerOnScreen();
+    }
+
+    /**
+     * Returns true if the note window is currently being shown.
+     */
+    public boolean isShowing() {
+        return getRoot().isShowing();
+    }
+
+    /**
+     * Hides the note window.
+     */
+    public void hide() {
+        getRoot().hide();
+    }
+
+    /**
+     * Focuses on the note window.
+     */
+    public void focus() {
+        getRoot().setIconified(false);
+        getRoot().requestFocus();
+    }
+
+    /**
+     * Saves the file
+     */
+    @FXML
+    public void handleSave() throws CommandException {
+        String paragraph = noteTextArea.getText();
+        Note editedNote = new Note(paragraph);
+        NoteDate newNoteDate = new NoteDate(DateUtil.getCurrentDateTime());
+        Person editedPerson = new Person(person.getName(), person.getPhone(), person.getEmail(),
+                editedNote, newNoteDate, person.getTags());
+        logic.executeSaveNote(person, editedPerson);
+    }
+
+
+    /**
+     * Exits the note window.
+     */
+    @FXML
+    public void handleExit() {
+        getRoot().close();
+        OPENED_NOTE_WINDOWS.remove(this);
+    }
+    /**
+     * Exits and saves the note window.
+     */
+    @FXML
+    public void handleSaveAndExit() throws CommandException {
+        handleSave();
+        handleExit();
+    }
+
+    /**
+     * Returns the index of first character of current line of the text cursor.
+     * @return index of current line
+     */
+    private int getLineIndex() {
+        int noOfCharsTillCursor = noteTextArea.getCaretPosition();
+        int wordIndex = 0;
+        ObservableList<CharSequence> noteCharSequence = noteTextArea.getParagraphs();
+        for (CharSequence charSequence : noteCharSequence) {
+            int noOfChars = charSequence.length() + 1;
+            if (wordIndex + noOfChars >= noOfCharsTillCursor) {
+                return wordIndex;
+            }
+            wordIndex += noOfChars;
+        }
+        return wordIndex;
+    }
+
+    private void timeStampNote() {
```

Comment:
Not sure, but given your description "Timestamps the note by inserting DateTime String with a line separator", I feel like it is intuitive to insert the line separator **underneath** and also put the cursor **underneath**. I think it's not very intuitive to be brought on top of what you are typing when you press Command T.

(context for those reading:
<img width="243" alt="Screenshot 2021-10-10 at 4 09 40 PM" src="https://user-images.githubusercontent.com/59471320/136687892-58483e6c-8b22-4c52-a2ae-5b62e5fa03db.png">

when you press command T at the end of your line of text, it prints the timestamp on top, and inserts a line separator **above that**, and then puts your cursor there above the date. I think it would make more sense to put the time stamp **where your caret is**. 

What do you guys think about the line separator? I'm wondering it it would be better not to have one.